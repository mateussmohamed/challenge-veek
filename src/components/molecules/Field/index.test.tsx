import React from 'react'

import { Field } from './index'
import withThemeProvider from '../../theme/withThemeProvider'

describe('[<Field />]', () => {
  const wrap = props => withThemeProvider(() => <Field {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({
      label: 'First Name',
      name: 'name',
    })

    expect(wrapper).toMatchSnapshot()
  })

  describe('when pass type prop', () => {
    test('matches to snapshot', () => {
      const wrapper = wrap({
        label: 'Upload your Avatar',
        name: 'avatar',
        type: 'file',
      })

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when pass multiline prop', () => {
    test('matches to snapshot with textarea', () => {
      const wrapper = wrap({
        label: 'Comments',
        name: 'comments',
        multiline: true,
      })

      expect(wrapper).toMatchSnapshot()
    })
  })
})
