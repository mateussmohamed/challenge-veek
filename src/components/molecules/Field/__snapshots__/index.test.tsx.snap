// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Field />] matches to snapshot 1`] = `[Function]`;

exports[`[<Field />] when pass multiline prop matches to snapshot with textarea 1`] = `[Function]`;

exports[`[<Field />] when pass type prop matches to snapshot 1`] = `[Function]`;
